typescript/packages/plugins/mode-governance/src/mode-governance.service.ts

```diff
@@ -0,0 +1,110 @@
+import { Tool } from "@goat-sdk/core";
+import { EVMWalletClient } from "@goat-sdk/wallet-evm";
+import { parseUnits, formatUnits } from "viem";
+import { VOTING_ESCROW_ABI } from "./abi";
+import {
+    MODE_VOTING_ESCROW,
+    BPT_VOTING_ESCROW,
+} from "./constants";
+import {
+    StakeParameters,
+    GetStakeInfoParameters,
+    GetBalanceParameters,
+} from "./parameters";
+
+export class ModeGovernanceService {
+    @Tool({
+        description: "Stake MODE or BPT tokens in the Mode governance system. Requires MODE or BPT tokens to be approved first.",
+    })
+    async stakeTokens(walletClient: EVMWalletClient, parameters: StakeParameters) {
+        const escrowAddress = parameters.tokenType === "MODE" ? MODE_VOTING_ESCROW : BPT_VOTING_ESCROW;
+        
+        const stakeHash = await walletClient.sendTransaction({
+            to: escrowAddress,
+            abi: VOTING_ESCROW_ABI,
+            functionName: "createLock",
+            args: [parseUnits(parameters.amount, 18)],
+        });
+
+        return stakeHash.hash;
+    }
+
+    @Tool({
+        description: "Get staking information including lock period and voting power",
+    })
+    async getStakeInfo(walletClient: EVMWalletClient, parameters: GetStakeInfoParameters) {
```

Comment:
Name is too generic, is not only staking but applies only to mode governance :)

maybe `get_mode_governance_stake_info`?

Reply:
sounds good